src/redux-middleware/pushQueue.ts

```diff
@@ -139,13 +133,54 @@ const flushPushQueue = (): Thunk<Promise<void>> => async (dispatch, getState) =>
 
   if (pushQueue.length === 0) return Promise.resolve()
 
-  dispatch(flushDeletes())
-  dispatch(flushEdits())
-  dispatch(flushMoves())
+  /**
+   * Returns updated thoughtIndexUpdates using the latest state after pulling pending lexemes.
+   */
+  const getSyncedThoughtsUpdate = (batch: PushBatch) => Object.keys(batch.thoughtIndexUpdates).reduce<Index<Lexeme | null>>((acc, thoughtId) => {
+
+    const updatedState = getState()
+
+    // Return early if this particular lexeme was not pending
+    const hadPendingLexeme = batch.pendingLexemes?.[thoughtId]
```

Comment:
Idea: Is it possible to calculate `batch.pendingLexemes` here instead of in `updateThoughts`? It isn't possible with `pendingMoves` for example which needs information from `existingThoughtMoves`, but `pendingLexemes` is a simple comparison with the state. One less dependency and one batch variable.

Reply:
I don't think so. `updateThoughts` updates the `thoughts` itself and we need to compare `batch` `thoughtIndexUpdates` against the current `thoughts`, not the updated one. In `pushQueue` we will already have the updated state so comparison will show no differences.